packages/react-reconciler/src/ReactFiberTreeReflection.js

```diff
@@ -117,16 +117,17 @@ export function findCurrentFiberUsingSlowPath(fiber: Fiber): Fiber | null {
   let b = alternate;
   while (true) {
     let parentA = a.return;
-    let parentB = parentA ? parentA.alternate : null;
-    if (!parentA || !parentB) {
+    if (!parentA) {
```

Comment:
I don’t think the logic is designed to make assumptions about which side is which. That’s why it’s a and b instead of say current and workInProgress.

This seems to think that A has some different property than B. Why is this?

Reply:
I believe the logic should have really been `!parentA && !parentB`. Then we should enter the next `if` if either is null. But it is impossible for parentA to be null and parentB to be non-null. Hence the asymmetry. I could write it in a more symmetric way but it would have dead branches.